server/auth/auth_oauth.py

```diff
@@ -121,10 +142,24 @@ def logout(self):
     def callback(self):
         token = self.client.authorize_access_token()
         id_token = token.get("id_token")
-        session[self.CXG_ID_TOKEN] = id_token
-        del session["oauth_callback_redirect"]
         oauth_callback_redirect = session.get("oauth_callback_redirect", "/")
+        del session["oauth_callback_redirect"]
```

Comment:
Can be more precisely refactored to:
`oauth_callback_redirect = session.pop("oauth_callback_redirect", "/")`

Reply:
fixed